### PR TITLE
Add '--logging-format' flag to kube-scheduler

### PR DESCRIPTION
**Pull request by SataQiu**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Add `--logging-format` flag to kube-scheduler

**Which issue(s) this PR fixes**:

Ref kubernetes/enhancements#1602

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
kube-scheduler: add '--logging-format' flag to support structured logging
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/component-base/logs/options.go [hunk 1]
```diff
@@ -55,8 +56,11 @@ func (o *Options) AddFlags(fs *pflag.FlagSet) {
 
 // Apply set klog logger from LogFormat type
 func (o *Options) Apply() {
-	// if log format not exists, use nil loggr
-	loggr, _ := o.Get()
+	var loggr logr.Logger
+	if o != nil {
+		// if log format not exists, use nil loggr
+		loggr, _ = o.Get()
+	}
```
Comment (serathius) [inline staging/src/k8s.io/component-base/logs/options.go:63]:
```suggestion
	if o != nil {
		return
	}
	loggr, _ := o.Get()
```

**Comment by ahg-g** [comment]
did you manually test that the flag now exist and takes an effect when set?

**Comment by lavalamp** [comment]
Like I said in the apiserver PR, I think we should wait on installing the flags until they have more than one option.

**Comment by serathius** [comment]
Json format was added. https://github.com/kubernetes/kubernetes/issues/91490
PR to update help listing available options is in flight. https://github.com/kubernetes/kubernetes/pull/92177

I think we can proceed with this PR. @lavalamp PTAL

**Comment by serathius** [comment]
PR updating flag was merged. Can go move forward with this one? @lavalamp

@SataQiu Please fix conflict

**Comment by lavalamp** [comment]
/hold

@liggitt asked for additional validation in another one of these. I think it might be overkill now that the flag is clearly labeled as alpha. But let's let that conversation finish first.

**Comment by liggitt** [comment]
I mostly want to make sure we don't open ourselves up to confused use of `--log-file=foo.log --logging-format=json` in a way which 1) doesn't work and 2) will change behavior in the future

**Comment by serathius** [comment]
@liggitt Flag validation was implemented, I tested it manually on apiserver. https://github.com/kubernetes/kubernetes/pull/92394#issuecomment-652046227
@lavalamp are we ok to move forward?

**Comment by serathius** [comment]
@lavalamp is OOO till Jul 6, this is to close to code freeze (Jul 9). 
@liggitt could you approve?

File: staging/src/k8s.io/component-base/logs/options.go [hunk 2]
```diff
@@ -67,6 +67,10 @@ func (o *Options) AddFlags(fs *pflag.FlagSet) {
 
 // Apply set klog logger from LogFormat type
 func (o *Options) Apply() {
+	if o == nil {
```
Comment (liggitt) [inline staging/src/k8s.io/component-base/logs/options.go:70]: guarding nil here and not in Validate() doesn't seem right... if o can be nil, won't we panic in Validate() as well?

Comment (serathius) [reply staging/src/k8s.io/component-base/logs/options.go:70]: I think was added to avoid adding defaults in options_tests.go. @SataQiu can you fix this?

Comment (liggitt) [reply staging/src/k8s.io/component-base/logs/options.go:70]: ah. yeah, let's avoid changing production code in confusing ways to accommodate unrealistic use by tests

Comment (SataQiu) [reply staging/src/k8s.io/component-base/logs/options.go:70]: OK,I will update it.

**Comment by serathius** [comment]
/lgtm
ping @ahg-g for approval of options_test.go

**Comment by SataQiu** [comment]
Could you have a look here?  thx~ @ahg-g

**Pull request merged**